packages/headless/doc-parser/src/entity-sorter.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  AnyEntity,
+  Entity,
+  EntityWithTypeAlias,
+  FuncEntity,
+  ObjEntity,
+} from './entity';
+
+export function sortEntities(entities: AnyEntity[]) {
+  const [methods, attributes] = divideIntoGroups(entities, (entity) =>
+    entity.kind === 'function' ? SortGroup.First : SortGroup.Second
```

Comment:
There is a `isFunctionEntity` utility that we could reuse here. It is a type-guard so typescript should infer the correct return type when true/false.